### PR TITLE
Add the iOS app examples and the sdk

### PR DESCRIPTION
**Pull request by houshengbo**
Closes-Bug: #502

### CODE REVIEW

File: tools/ubuntu-setup/ansible.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo apt-get install -y software-properties-common
 sudo apt-add-repository -y ppa:ansible/ansible
 sudo apt-get update
-sudo apt-get install -y ansible=2.0.2.0-1ppa~trusty
+sudo apt-get install -y ansible
```
Comment (markusthoemmes) [inline tools/ubuntu-setup/ansible.sh]: Why make that change here? Bad push?

**Comment by houshengbo** [comment]
This pull request is ready for review now.

File: build.xml [hunk 2]
```diff
@@ -39,6 +39,7 @@
             <arg value="-PdockerHost=${main.docker.endpoint}" />
         </exec>
         <ant antfile="${openwhisk.home}/services/nginx/build.xml" target="buildProxy" />
+    	<ant antfile="tools/build.xml" target="build" />
```
Comment (rabbah) [inline build.xml]: fix indentation.

File: build.xml [hunk 3]
```diff
@@ -52,7 +53,7 @@
         <ant antfile="${openwhisk.home}/services/nginx/build.xml" target="startProxy" />
         <ant antfile="services/build.xml" target="deploy" />
         <ant antfile="core/build.xml" target="deploy" />
-        <ant antfile="tools/cli/build.xml" target="buildCLI" />
+    	<ant antfile="tools/build.xml" target="deploy" />
```
Comment (rabbah) [inline build.xml]: fix indentation.

File: tools/build.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<!-- vim: set expandtab ts=4 sw=4 : -->
+
+<!-- targets in this file should be called from the parent directory -->
+<project basedir="..">
+	<import file="../config/config.xml" />
```
Comment (rabbah) [inline tools/build.xml]: fix indentation.

**Comment by markusthoemmes** [comment]
We're not using `ant` for builds anymore. Going forward, all builds should be based on `gradle`. Is that possible here?

**Comment by houshengbo** [comment]
@markusthoemmes, do we get rid of all the ant? 
How is the gradle doing in OpenWhisk. Are we able to deploy OpenWhisk with gradle? Does it document somewhere?
Thank you.

**Comment by rabbah** [comment]
ant is just a shell for gradle right now - most of the components are built with gradle already with the exception of this and the CLI (there might be one more). If you look at the ant targets for the core components, they call gradle. Here is just one example: https://github.com/openwhisk/openwhisk/blob/master/build.xml#L36

**Comment by lehoanganh** [comment]
deploy with ansible is happening, ant is soon or later obsolete

**Comment by houshengbo** [comment]
What is the issue number for adding the deployment with ansible?
Who is the assignee? Is there any pull request in progress, so that folks are able to see?

**Comment by csantanapr** [comment]
rebase and merge 75f6615e37b54a7a9d01272c2cac60570fb787b2

**Pull request closed without merging**